be/src/olap/types.cpp

```diff
@@ -44,27 +44,24 @@ class ScalarTypeInfoResolver {
     DECLARE_SINGLETON(ScalarTypeInfoResolver);
 
 public:
-    std::shared_ptr<const TypeInfo> get_type_info(const FieldType t) {
-        auto pair = _scalar_type_mapping.find(t);
-        DCHECK(pair != _scalar_type_mapping.end()) << "Bad field type: " << t;
-        return pair->second;
+    std::shared_ptr<const TypeInfo> get_type_info(const FieldType t) const {
+        return _scalar_type_mapping[t];
```

Comment:
how about use template here？
so we don't keep OLAP_FIELD_TYPE_SIZE and _scalar_type_mapping？

Reply:
Sounds like it can be returned at compile time?

Reply:
It should be that template functions are specialized for each type at compile time

Reply:
>It should be that template functions are specialized for each type at compile time
+1